src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,117 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+
+    private final String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads tasks from the specified filePath
+     *
+     * @return An ArrayList of Tasks loaded from the file.
+     * @throws DukeException throws exception when Tasks cannot be loaded
+     */
+    public ArrayList<Task> loadTasks() throws DukeException {
+        try {
+            ArrayList<Task> res = new ArrayList<>();
+            File taskFile = new File(this.filePath);
+            if (!taskFile.createNewFile()) {
+                Scanner fileReader = new Scanner(taskFile);
+                while (fileReader.hasNextLine()) {
+                    String data = fileReader.nextLine();
+                    res.add(dataToTask(data));
+                }
+            }
+            return res;
+        } catch (IOException | DukeException e) {
+            throw new DukeException("Unable to load tasks from file.");
+        }
+    }
+
+    private static Task dataToTask(String str) throws DukeException {
+        String[] taskArr = str.split(",");
+        String taskType = taskArr[0];
+        boolean taskDone = taskArr[1].equals("1");
+        String taskDescription = taskArr[2];
+        String taskDate = "";
+        if (taskArr.length > 3) {
+            taskDate = taskArr[3];
+        }
+        Task res = null;
+        switch (taskType) {
+            case("T"):
+                res = new Todo(taskDescription, taskDone);
+                break;
+            case("D"):
+                res = new Deadline(taskDescription, taskDate, taskDone);
+                break;
+            case("E"):
+                res = new Event(taskDescription, taskDate, taskDone);
+                break;
+        }
```

Comment:
Just my opinion, but maybe a `default` block in `switch-case` to catch unknown edge cases would be nice if possible.